src/quickunlock/WindowsHello.cpp

```diff
@@ -97,23 +97,10 @@ namespace
     }
 } // namespace
 
-WindowsHello* WindowsHello::m_instance{nullptr};
-WindowsHello* WindowsHello::instance()
-{
-    if (!m_instance) {
-        m_instance = new WindowsHello();
-    }
-    return m_instance;
-}
-
 WindowsHello::WindowsHello(QObject* parent)
     : QObject(parent)
 {
-    concurrency::create_task([this] {
-        bool state = KeyCredentialManager::IsSupportedAsync().get();
-        m_available = state;
-        emit availableChanged(m_available);
-    });
+    m_available = KeyCredentialManager::IsSupportedAsync().get();
```

Comment:
Would be ideal if this was sync - if someone can test on Windows that would be great

Reply:
I [looked it up](https://learn.microsoft.com/en-us/uwp/api/windows.security.credentials.keycredentialmanager.issupportedasync?view=winrt-22621#windows-security-credentials-keycredentialmanager-issupportedasync) and initially I didn't see how the asynchronicity works here, since there's apparently no callback. But after digging a little deeper, I see the `.get()` call [makes it synchronous](https://learn.microsoft.com/en-us/uwp/api/windows.foundation.iasyncoperation-1?view=winrt-22621#cwinrt-extension-functions). So it should work.

Reply:
Why was windowshello code touched at all?

Reply:
I suppose to get it in line with the new common interface. The previous code was using a signal that's not part of the interface.

Technically, this could just skip the `emit` call, but after a skim of the API docs, I think the new code is enough.

Reply:
The .get() could cause the ui to freeze if windows takes a while to process the request.